addons/repair/models/repair.py

```diff
@@ -537,12 +537,19 @@ def action_validate(self):
         }
 
     def action_view_sale_order(self):
-        return {
+        sale_order_ids = self.mapped('sale_order_id.id')
+        action_window = {
             "type": "ir.actions.act_window",
             "res_model": "sale.order",
-            "views": [[False, "form"]],
-            "res_id": self.sale_order_id.id,
+            "name": _("Sales Order"),
         }
+        if len(sale_order_ids) == 1:
+            action_window["views"] = [[False, "form"]]
+            action_window["res_id"] = sale_order_ids[0]
+        else:
+            action_window["views"] = [[False, "tree"], [False, "form"]]
+            action_window["domain"] = [["id", "in", sale_order_ids]]
+        return action_window
```

Comment:
Cleanup nitpicks:

- Use `self.sale_order_id.ids` instead of `self.mapped('sale_order_id.id')`
- Name it `action` instead of `action_window` to follow the common naming of actions
- Use `action.update({dict_values})` instead of of `action_window["views"]` to follow the [standard](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#idiomatics-of-programming-python)
- if you want `"Sales Order"` to be translated, you need to re-export the pot file
- Within logistics code, we tend to prefer using `' '` for code and `" "` for user facing strings
- note that in 18.0+, `"tree"` is replaced with `"list"`